### PR TITLE
Mapbox access token catches

### PR DESCRIPTION
**Pull request by etpinard**
Fixes bug introduced in https://github.com/plotly/plotly.js/pull/729 and on display here: http://codepen.io/lpan/pen/kXJzbY?editors=0011 discovered by @lpan .

```js
Plotly.plot(gd, data, layout).catch(function(err) {
  console.log('works again');
});
```

### CODE REVIEW

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -326,10 +326,6 @@ Plotly.plot = function(gd, data, layout, config) {
     // so that the caller doesn't care which route we took
     return Promise.all(gd._promises).then(function() {
         return gd;
-    }, function() {
-        // clear the promise queue if one of them got rejected
-        Lib.log('Clearing previous rejected promises from queue.');
-        gd._promises = [];
```
Comment (etpinard) [inline src/plot_api/plot_api.js]: I was under the impression that the 2nd argument to `Promise.all().then(, )` was simply a callback and **did not* eat the any subsequent `Plotly.plot().catch()` handlers. I was wrong.

File: src/plot_api/plot_api.js [hunk 2]
```diff
@@ -1536,6 +1538,7 @@ Plotly.moveTraces = function moveTraces(gd, currentIndices, newIndices) {
 // style files that want to specify cyclical default values).
 Plotly.restyle = function restyle(gd, astr, val, traces) {
     gd = getGraphDiv(gd);
+    clearPromiseQueue(gd);
```
Comment (etpinard) [inline src/plot_api/plot_api.js]: Now, clear the `gd._promises` on `restyle` and `relayout` instead.

File: test/jasmine/tests/mapbox_test.js [hunk 3]
```diff
@@ -180,18 +180,26 @@ describe('mapbox credentials', function() {
...
+
         Plotly.plot(gd, [{
             type: 'scattermapbox',
             lon: [10, 20, 30],
             lat: [10, 20, 30]
         }], {}, {
             mapboxAccessToken: dummyToken
         }).catch(function(err) {
+            cnt++;
             expect(err).toEqual(new Error(constants.mapOnErrorMsg));
```
Comment (etpinard) [inline test/jasmine/tests/mapbox_test.js]: This assertion :arrow_double_up: did not get evaluated before that is this `it` would didn't test anything.

I wish jasmine had the something similar to TAP's `t.plan`.

**Comment by lpan** [comment]
Error caught in filewell! Thanks a lot man :+1: :dancer:

**Pull request merged**